src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -80,6 +94,10 @@
                 (! result.containsKey("accept-encoding"))) {
             result.put("accept-encoding", new BasicHeader("Accept-Encoding", "gzip, deflate"));
         }
+        if (options.getCompressRequestBody() == CompressType.GZIP &&
+                (! result.containsKey("content-encoding"))) {
+            result.put("content-encoding", new BasicHeader("Content-Encoding", "gzip"));
+        }
```

Comment:
Yeah, it might cause a problem.  Depends upon whether the client and server might be doing something "custom" which would allow the server to still end up decoding the request body properly.  I figured it would be safer to follow the pattern we used previously for other headers - like with `accept-encoding` above - where if the client specified a value for the header that we should just trust that the client wants that value to go out verbatim in the request rather than overriding the client's intent.  Does that seem reasonable?